core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -185,39 +197,33 @@ public TenantObject setTrustAnchor(final X509Certificate certificate) {
      * value as the <em>printable</em> form but without the leading
      * {@code -----BEGIN CERTIFICATE-----} and trailing {@code -----END CERTIFICATE-----}.
      * 
+     * @deprecated Replaced by {@link #getTrustedCertificateAuthorities()}
+     * 
      * @return The certificate or {@code null} if no certificate authority
      *         has been set.
      * @throws CertificateException if the value of <em>cert</em> property cannot be parsed into
-     *             an X.509 certificate.
+     *              an X.509 certificate.
      */
+    @Deprecated(since="1.0", forRemoval=true)
```

Comment:
I do not think that we should deprecate this method. It is only used internally for deserializing a trust anchor that has been specified in the form of an X.509 certificate. Thus, I guess we should make it private and change its signature to accept a JsonObject that contains the cert.

Reply:
> Thus, I guess we should make it private and change its signature to accept a JsonObject that contains the cert.

yes, such a method already exist. 

Like i said before, the reason why i deprecated it for now is so that I do not touch the *-adapters code base in this PR. i will remove it entirely in a follow up PR